### ISSUE TITLE
Create a Challenge System.

### ISSUE THREAD

**Issue by kantagara** [ChainSafe/web3.unity#764]

**Comment by oleksandrchainsafe** [comment]
Have to refresh my knowledge of the game design first, but overall sounds reasonable. Reminds me of how achievements are usually implemented.
I suppose there is a set of challenges for the current session that have to be completed in order to finish the level, right?

**Comment by kantagara** [comment]
Yes, pool of challenges are going to be present and two of them are going
to be taken per session.
